repaygithub/ui-tools
# CACTUS-848 :: Dependency Upgrades

Ticket: https://repayonline.atlassian.net/browse/CACTUS-848

I'm pretty sure the latest version of `@repay/testing-tools` on npm is broken. It was missing its `esm` dependency, but I added it back here.

### Testing

I was having issues when I tried to use this version of `@repay/scripts` in cactus, which I believe to be a problem in cactus itself (thread [here](https://repay.slack.com/archives/CABGTU4S3/p1645562577572169) if you haven't already seen it), so you'll have to test the dev command using an app generated with `create-repay-ui`. We can still test everything else in cactus though.

1. Pull down this branch and `yarn install`
2. `cd` into each package and `yalc publish` for each of them.
3. Move over to the cactus repo, and run `yalc add --dev @repay/testing-tools` at the root, and then `yarn install`.
4. `cd examples/mock-ebpp`, `yarn test:local` and make sure the tests pass.
5. Now go back to the root and run `yalc add --dev @repay/babel-preset` and then `yarn install`
6. Run `yarn build` at the root and make sure all of the packages build successfully.
7. Run `yalc add --dev @repay/eslint-config` at the root and then `yarn install`
8. Run `yarn lint` at the root and make sure linting works
9. Run `yalc add --dev @repay/scripts` at the root and then `yarn install`
10. Run `yarn build` at the root and make sure everything builds
11. Now `cd examples/mock-ebpp` and run `yarn build`. Make sure that app builds.
12. Now, generate a new app using our CLI. After it's done generating, `cd` into the project and run `yalc add --dev @repay/scripts`, followed by `yan install`
13. Run `yarn start` in the generated app and make sure it spins up and you can view it in the browser
